public/app/core/components/AppChrome/AppChrome.tsx

```diff
@@ -61,18 +79,50 @@ export function AppChrome({ children }: Props) {
           </div>
         </>
       )}
-      <main className={contentClass} id="pageContent">
+      <main className={contentClass}>
         <div className={styles.panes}>
           {state.layout === PageLayoutType.Standard && state.sectionNav && !config.featureToggles.dockedMegaMenu && (
             <SectionNav model={state.sectionNav} />
           )}
-          <div className={styles.pageContainer}>{children}</div>
+          <div className={styles.pageContainer} id="pageContent">
+            {children}
+          </div>
         </div>
       </main>
       {!state.chromeless && (
         <>
           {config.featureToggles.dockedMegaMenu ? (
-            <DockedMegaMenu searchBarHidden={searchBarHidden} onClose={() => chrome.setMegaMenu(false)} />
+            <div className={styles.menuWrapper}>
```

Comment:
Can you maybe explain a bit more about the motivation/why you did this refactor, what the goals were, what's better about this compared to before?

This PR doubles the size of this file/component, mostly with animation boilerplate.

Without hearing what the purpose of this refactor was, I think it might be better to split the backdrop + animation out into a seperate component rather than having all of that in here?

Reply:
yeah sure. before, the focus trapping, animation and overlay behaviour were all tied to `DockedMegaMenu`. this makes it so that `DockedMegaMenu` is now a standalone component that can be used in both contexts, either within `AppChrome` as a slide out menu or inline as a docked menu. 

yeah it might be worth creating something like `AppChromeMenu`, which basically just wraps all this code up and encapsulates it away? 🤔

Reply:
I thought either you could move this out into something like `AppChromeMenu` that abstracts all of it.

or, could do something like

```ts
<AppChromeMenuPresenter>
  <DockedMegaMenu ... />
</AppChromeMenuPresenter>
```

which owns backdrop + transitions, but renders the menu of the children.